royalties-handler/src/shareholders.rs

```diff
@@ -0,0 +1,91 @@
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+use crate::reward_entries::FIRST_ENTRY_ID;
+
+#[elrond_wasm::module]
+pub trait ShareholdersModule:
+    crate::common_storage::CommonStorageModule
+    + crate::reward_entries::RewardEntriesModule
+    + crate::token_balance::TokenBalanceModule
+{
+    #[only_owner]
+    #[endpoint(addShareholders)]
+    fn add_shareholders(&self, #[var_args] shareholders: MultiValueEncoded<ManagedAddress>) {
+        let mut mapper = self.shareholders();
+        for sh in shareholders {
+            let _ = mapper.insert(sh);
+        }
+    }
+
+    #[only_owner]
+    #[endpoint(removeShareholders)]
+    fn remove_shareholders(&self, #[var_args] shareholders: MultiValueEncoded<ManagedAddress>) {
+        let mut mapper = self.shareholders();
+        for sh in shareholders {
+            let _ = mapper.swap_remove(&sh);
+        }
+    }
+
+    #[endpoint(claimRewards)]
+    fn claim_rewards(&self, #[var_args] entry_ids: MultiValueEncoded<usize>) {
```

Comment:
Why give entry_ids as an argument and not just collect all available rewards for the caller by default? It adds an extra off-chain computation step and is more prone to errors.

Reply:
The problem is that if a user does not claim for a long time, this would always run out of gas. These IDs can easily be retrieved through `get_claimable_entry_ids_for_address`, which maybe the front-end could then batch to not run into gas issues.